internal/distributed/proxy/httpserver/handler_v1.go

```diff
@@ -0,0 +1,582 @@
+	} else {
+		c.JSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusOK, HTTPReturnData: gin.H{}})
+	}
+}
+
+func (h *Handlers) query(c *gin.Context) {
+	httpReq := QueryReq{
+		DbName: DefaultDbName,
+		Limit:  100,
+	}
+	if err := c.ShouldBindBodyWith(&httpReq, binding.JSON); err != nil {
+		c.AbortWithStatusJSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusBadRequest, HTTPReturnMessage: "check your parameters conform to the json format", HTTPReturnError: err.Error()})
+		return
+	}
+	if httpReq.CollectionName == "" {
+		c.AbortWithStatusJSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusBadRequest, HTTPReturnMessage: "collectionName is required."})
+		return
+	}
+	coll, err := h.describeCollection(c, httpReq.DbName, httpReq.CollectionName, false)
+	if err != nil || coll == nil {
+		return
+	}
+	var dynamicFields []string
+	httpReq.OutputFields, dynamicFields = getDynamicOutputFields(coll.Schema, httpReq.OutputFields)
+	req := milvuspb.QueryRequest{
+		CollectionName:     httpReq.CollectionName,
+		Expr:               httpReq.Filter,
+		OutputFields:       httpReq.OutputFields,
+		GuaranteeTimestamp: BoundedTimestamp,
+	}
+	username, _ := c.Get(ContextUsername)
+	_, authErr := proxy.PrivilegeInterceptorWithUsername(c, username.(string), &req)
+	if authErr != nil {
+		c.JSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusUnauthorized, HTTPReturnMessage: authErr.Error()})
+		return
+	}
+	response, err := h.proxy.Query(c, &req)
+	if err != nil {
+		c.JSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusBadRequest, HTTPReturnMessage: "query fail", HTTPReturnError: err.Error()})
+	} else if response.Status.ErrorCode != commonpb.ErrorCode_Success {
+		c.JSON(http.StatusOK, gin.H{HTTPReturnCode: response.Status.ErrorCode, HTTPReturnMessage: response.Status.Reason})
+	} else {
+		outputData, err := buildQueryResp(int64(0), dynamicFields, response.FieldsData, nil)
+		if err != nil {
+			c.JSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusBadRequest, HTTPReturnMessage: "show result by row wrong", "originData": response.FieldsData, HTTPReturnError: err.Error()})
+		} else {
+			c.JSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusOK, HTTPReturnData: outputData})
+		}
+	}
+}
+
+func (h *Handlers) get(c *gin.Context) {
+	httpReq := GetReq{
+		DbName: DefaultDbName,
+	}
+	if err := c.ShouldBindBodyWith(&httpReq, binding.JSON); err != nil {
+		c.AbortWithStatusJSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusBadRequest, HTTPReturnMessage: "check your parameters conform to the json format", HTTPReturnError: err.Error()})
+		return
+	}
+	if httpReq.CollectionName == "" {
+		c.AbortWithStatusJSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusBadRequest, HTTPReturnMessage: "collectionName is required."})
+		return
+	}
+	coll, err := h.describeCollection(c, httpReq.DbName, httpReq.CollectionName, false)
```

Comment:
Describing the Collection is unnecessary; the Result contains all the data you require.

Reply:
`/vector/get` request only provide the parameter `id`, but the primary field may be not named `id`.